docs/package-overview.md

```diff
@@ -0,0 +1,49 @@
+# Overview of PHP-HTTP packages
+
+All packages are installed via [Composer](http://getcomposer.org/).
+
+
+## httplug
+
+**Interfaces for HTTP Client**
+
+- Namespace: `Http\Client`
+- Repository: https://github.com/php-http/httplug
+- Documentation: (Httplug)[httplug.md]
+
+
+## message-factory
+
+**Abstraction to create PSR-7 requests without binding to a specific implementation**
+ 
+- Namespace: `Http\Message`
+- Repository: https://github.com/php-http/message-factory
+- Documentation: [Message Factory](message-factory.md) 
+
+
+## *-client, *-adapter
+
+**Each client implementation is separated into its own package. This allows for clean requirement specification.**
+
+- Namespace: `Http\Client`
```

Comment:
Sure conflicts in autoloading if you have many packages installed at the same time that configure the same namespaces means that the autoloader will have to look into a few directories before it finds a match for a class, so that might not be good, but I have zero context as to what you are doing here so not sure if it's really a problem or not.